pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentSubscription.java

```diff
@@ -686,12 +839,46 @@ public SubscriptionStats getStats() {
 
     @Override
     public synchronized void redeliverUnacknowledgedMessages(Consumer consumer) {
-        dispatcher.redeliverUnacknowledgedMessages(consumer);
+        ConcurrentLongLongPairHashMap positionMap = consumer.getPendingAcks();
+        // Check if message is in pending_ack status.
+        if (null != positionMap) {
+            List<PositionImpl> pendingPositions = new ArrayList<>();
```

Comment:
same comment as what I made in the other `redeliverUnacknowledgedMessages` method.